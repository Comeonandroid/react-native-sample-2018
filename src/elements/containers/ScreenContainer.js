import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import { View } from '../index'

const ScreenContainer = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

ScreenContainer.propTypes = {
  children: PropTypes.node
}

const Element = styled(View) `
  position: relative;
  flex: 1;
  display: flex;
  background-color: ${ ({ theme }) => theme.colors.background };
`

export default ScreenContainer
