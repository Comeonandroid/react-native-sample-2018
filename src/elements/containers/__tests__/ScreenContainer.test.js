import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import ScreenContainer from '../ScreenContainer'

describe('<ScreenContainer />', () => {
  itBehavesLike('aThemedComponent', { subject: <ScreenContainer /> })
  itBehavesLike('aComponentWithChildren', { subject: <ScreenContainer /> })
})
