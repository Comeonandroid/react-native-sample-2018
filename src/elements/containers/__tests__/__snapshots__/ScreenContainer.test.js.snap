// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenContainer /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "backgroundColor": "#000000",
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "position": "relative",
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<ScreenContainer /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "backgroundColor": "#fefaec",
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "position": "relative",
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<ScreenContainer /> renders component when passed in 1`] = `
<Styled(View)>
  <Component />
</Styled(View)>
`;

exports[`<ScreenContainer /> renders text when passed in 1`] = `
<Styled(View)>
  Test
</Styled(View)>
`;
